matrix_sdk_base/src/store/sled_store/mod.rs

```diff
@@ -781,6 +781,24 @@ impl SledStore {
 
         Ok(self.media.apply_batch(batch)?)
     }
+
+    async fn remove_room(&self, room_id: &RoomId) -> Result<()> {
+        self.members.remove(&room_id.as_bytes())?;
```

Comment:
Are these correct? I don' think they are, you need to pass in the full key, the `room_id` will only be the first part of the key. A test should confirm this.

For example, for the `members` tree this will be the `room_id` + `null byte` + `state_key` + `null byte`.

I think you'll need to use [`scan_prefix()`](https://docs.rs/sled/0.34.6/sled/struct.Db.html#method.scan_prefix) to get the full keys and then remove each of the keys.

Another thing, this should be done in a transaction so all the room info gets removed or none of it, so we don't end up in an inconsistent state where only half of the room info is missing.